docs/user-manual/index.md

```diff
@@ -1660,6 +1693,103 @@ reaches the rate limit. It is normal to see messages in the
+
+The contents of the private key file is used as the value of `umapi_private_key_data` in the credential store.
+
+The credential values will be looked up using the specified key names with the user being the org_id value.
+
+
+connector-ldap.yml
+
+	username: "your ldap account username"
+	secure_password_key: ldap_password 
+	host: "ldap://ldap server name"
+	base_dn: "DC=domain name,DC=com"
+
+The LDAP access password will be looked up using the specified key name
+(`ldap_password` in this example) with the user being the specified username
+config value.
+
+Credentials are stored in the underlying operating system secure store.  The specific storage system depends in the operating system.
+
+| OS | Credential Store |
+|------------|--------------|
+|Windows | Windows Credential Vault |
+| Mac OS X | Keychain |
+| Linux | Freedesktop Secret Service or KWallet |
+
+On Linux, the secure storage application would have been installed and configured by the OS vendor.
+
+The credentials are added to the OS secure storage and given the username and credential id that you will use to specify the credential.  For umapi credentials, the username is the organization id.  For the LDAP password credential, the username is the LDAP username.  You can pick any identifier you wish for the specific credentials; they must match between what is in the credential store and the name used in the configuration file.  Suggested values for the key names are shown in the examples above.
+
+
+#### Storing Credential Files in External Management Systems
+
+As an alternative to storing credentials in the local credential store, it is possible to integrate User Sync with some other system or encryption mechanism.  To support such integrations, it is possible to store the entire configuration files for umapi and ldap externally in some other system or format.
+
+This is done by specifying, in the main User Sync configuration file, a command to be executed whose output is used as the umapi or ldap configuration file contents.  You will need to provide the command that fetches the configuration information and sends it to standard output in yaml format, matching what the configuration file would have contained.
+
+To set this up, use the following items in the main configuration file.
+
+
+user-sync-config.yml (showing partial file only)
+
+	adobe_users:
+	   connectors:
+	      # umapi: connector-umapi.yml   # instead of this file reference, use:
+	      umapi: $(read_umapi_config_from_s3)
+	      # if a working directory is required:
+	      # umapi $([temp]read_umapi_config_from_s3) # runs command in "temp" folder
```

Comment:
Actually, there is no provision for the command to specify a different working directory: it is always run in the same directory as the configuration file.  If they want to invoke the command somewhere else, they need to specify the full path to the command as part of the command.  And if the command needs a different working directory, it needs to change to that working directory itself, and take it as an argument.  (I can see you were reading the code, but that `[...]` convention is not used on input.)  Do you think we really need to be able to specify a different working directory?  if you do I can easily make this description true. :)